lib/node/runner/api_node.go

```diff
@@ -124,10 +125,29 @@ func (api NetworkHandlerNode) MessageHandler(w http.ResponseWriter, r *http.Requ
 		Conf:            api.conf,
 	}
 
-	if err = common.RunChecker(checker, common.DefaultDeferFunc); err != nil {
+	err = common.RunChecker(checker, common.DefaultDeferFunc)
+
+	if err != nil {
 		if len(checker.Transaction.H.Hash) > 0 {
 			block.SaveTransactionHistory(api.storage, checker.Transaction, block.TransactionHistoryStatusRejected)
 		}
+		return
+	}
+
+	tx = checker.Transaction
+	return
```

Comment:
How about returning with variable like `return checker.Transaction, nil`?
It looks more clear.